src/twisted/conch/client/knownhosts.py

```diff
@@ -431,7 +431,7 @@ def hasHostKey(self, hostname, key):
             does not match the given key.
         """
         for lineidx, entry in enumerate(self.iterentries(), -len(self._added)):
-            if entry.matchesHost(hostname):
+            if entry.matchesHost(hostname) and entry.keyType == key.sshType():
```

Comment:
This feels like a separate fix that should have been addressed in a different ticket?